Sources/String+Template.swift

```diff
@@ -176,15 +176,25 @@ extension String {
 
         var formatted = template.template
         var currentIndex = startIndex;
-        var formatStringIndex = formatted.index(of: template.placeholderToken)
+        var formatStringIndex: String.Index?
+        #if swift(>=5)
+            formatStringIndex = formatted.firstIndex(of: template.placeholderToken)
```

Comment:
Thoughts on polyfilling this instead? Basically define `firstIndex(of:)` for Swift < 5.0 in a different file so that we don't litter this code with version-specific changes?

Reply:
Turns out `firstIndex(of:)` is available in Swift 4 (and maybe before), so no need to polyfill or add conditionals.